File: hackertracker/src/main/java/com/shortstack/hackertracker/List/ScheduleItemBuilder.java
package com.shortstack.hackertracker.List;

import com.pedrogomez.renderers.RendererBuilder;
import com.shortstack.hackertracker.Model.Item;
import com.shortstack.hackertracker.Renderer.GenericHeaderRenderer;
import com.shortstack.hackertracker.Renderer.RelativeTimeRenderer;
import com.shortstack.hackertracker.Renderer.ItemRenderer;

import java.util.Date;


public class ScheduleItemBuilder extends RendererBuilder<Item> {

    public ScheduleItemBuilder() {

        bind(Item.class, new ItemRenderer())
                .bind(String.class, new GenericHeaderRenderer())
                .bind(Date.class, new RelativeTimeRenderer());
    }
}

File: hackertracker/src/main/java/com/shortstack/hackertracker/List/ScheduleItemBuilder.kt
package com.shortstack.hackertracker.List

import com.pedrogomez.renderers.RendererBuilder
import com.shortstack.hackertracker.Model.Day
import com.shortstack.hackertracker.Model.Item
import com.shortstack.hackertracker.Model.Time
import com.shortstack.hackertracker.Renderer.ItemRenderer
import com.shortstack.hackertracker.Renderer.RelativeDayRender
import com.shortstack.hackertracker.Renderer.RelativeTimeRenderer


class ScheduleItemBuilder : RendererBuilder<Item>() {
    init {

        bind(Item::class.java, ItemRenderer())
                .bind(Day::class.java, RelativeDayRender())
                .bind(Time::class.java, RelativeTimeRenderer())
    }
}
